File: apps/web/src/types/docs.ts
export type DocumentationLanguage = "curl" | "python" | "javascript" | "nodejs";

export interface DocumentationCodeExample {
  language: DocumentationLanguage;
  label: string;
  code: string;
}

export interface DocumentationSection {
  title: string;
  body: string;
}

export interface ToolDocumentation {
  tool_id: string;
  tool_slug: string;
  tool_name: string;
  endpoint_url: string;
  method: string;
  authentication: DocumentationSection;
  request_format: DocumentationSection;
  response_format: DocumentationSection;
  rate_limit: DocumentationSection;
  error_codes: Array<{
    status: number;
    code: string;
    meaning: string;
  }>;
  request_example: Record<string, unknown> | unknown[] | string;
  response_example: Record<string, unknown> | unknown[] | string;
  code_examples: DocumentationCodeExample[];
}

File: apps/api/app/schemas/docs.py
import uuid
from typing import Literal

from pydantic import BaseModel


class DocumentationCodeExample(BaseModel):
    language: Literal["curl", "python", "javascript", "nodejs"]
    label: str
    code: str


class DocumentationSection(BaseModel):
    title: str
    body: str


class ToolDocumentation(BaseModel):
    tool_id: uuid.UUID
    tool_slug: str
    tool_name: str
    endpoint_url: str
    method: str
    authentication: DocumentationSection
    request_format: DocumentationSection
    response_format: DocumentationSection
    rate_limit: DocumentationSection
    error_codes: list[dict[str, str | int]]
    request_example: dict | list | str
    response_example: dict | list | str
    code_examples: list[DocumentationCodeExample]
